### ISSUE TITLE
Invalid option for mac os

### ISSUE THREAD

**Issue by ajainvivek** [pedro-hs/checkbox.sh#6]
Looks like the script doesn't work on Mac OS 

I am receiving error ->

`
./scripts/utils/checkbox.sh: line 416: read: -N: invalid option
read: usage: read [-ers] [-u fd] [-t timeout] [-p prompt] [-a array] [-n nchars] [-d delim] [name ...]
`

**Comment by pedro-hs** [comment]
Unfortunately i don't have access to a Mac

Seeing the error it looks like it happened on one of those lines
```
IFS= read -sN1 key 2>/dev/null >&2
read -sN1 -t 0.0001 k1
read -sN1 -t 0.0001 k2
read -sN1 -t 0.0001 k3
```

In my analysis i'm considering that the error is on one of those lines

Comparing the docs of the command `read` on:
- linux: https://ss64.com/osx/read.html
- mac: https://linuxcommand.org/lc3_man_pages/readh.html

I can see mac don't have the `-N` option
- Description of the option on linux: `-N` nchars return only after reading exactly NCHARS characters, unless EOF is encountered or read times out, ignoring any delimiter

But both OS have the `-n` option,
- Description of the option on mac: `-n` nchars read returns after reading nchars characters rather than waiting for a complete line of input
- Description of the option on linux: `-n` nchars return after reading NCHARS characters rather than waiting for a newline, but honor a delimiter if fewer than NCHARS characters are read before the delimiter

Maybe change `-sN1` to `-sn1` solve the problem
I think this won't break anything in the script because the delimiter option (-d) of the command `read` is not used on this script

Try change this code:
```
IFS= read -sN1 key 2>/dev/null >&2
read -sN1 -t 0.0001 k1
read -sN1 -t 0.0001 k2
read -sN1 -t 0.0001 k3
```

For this:
```
IFS= read -sn1 key 2>/dev/null >&2
read -sn1 -t 0.0001 k1
read -sn1 -t 0.0001 k2
read -sn1 -t 0.0001 k3
```

In my tests, the script worked well
OS: Ubuntu 20
Terminal: Terminator
Shell: Bash

If this solve your problem i'll make a pull request with this change

**Comment by pedro-hs** [comment]
I merged the possible solution on master
Like i said: `I think this won't break anything in the script because the delimiter option (-d) of the command read is not used on this script`
So don't need to use the N option

**Comment by andreysmyntyna** [comment]
Hi,

This is how it looks like now on MacOs Catalina

```
\e[2K\e[37m--------------------------------------------------
\e[2K\e[34m    [ ] Option 1
\e[2K\e[37m    [ ] Option 2
\e[2K\e[37m    [ ] Option 3
\e[2K\e[37m    [ ] Option 4
\e[2K\e[37m    [ ] Option 5
\e[2K\e[37m    [ ] Option 6
\e[2K\e[37m    [ ] Option 7
\e[2K\e[37m    [ ] Option 8
\e[2K\e[37m    [ ] Option 9
\e[2K\e[37m    [ ] Option 10
\e[2K\e[37m    [ ] Option 11
\e[2K\e[37m    [ ] Option 12
\e[2K\e[37m    [ ] Option 13
\e[2K\e[37m    [ ] Option 14
\e[2K\e[37m    [ ] Option 15
\e[2K\e[37m    [ ] Option 16
\e[2K\e[37m    [ ] Option 17
\e[2K\e[37m    [ ] Option 18
\e[2K\e[37m    [ ] Option 19
\e[2K\e[37m    [ ] Option 20
\e[2K\e[37m    [ ] Option 21
\e[2K\e[37m    [ ] Option 22
\e[2K\e[37m    [ ] Option 23
\e[2K\e[37m    [ ] Option 24
\e[2K\e[37m    [ ] Option 25
\e[2K\e[37m    [ ] Option 26
\e[2K\e[37m    [ ] Option 27
\e[2K\e[37m    [ ] Option 28
\e[2K\e[37m    [ ] Option 29
\e[2K\e[37m    [ ] Option 30
\e[2K\e[37m--------------------------------------------------
  1/30

```

**Comment by pedro-hs** [comment]
@andreysmyntyna thanks for the feedback,
I'll search more about ANSI colors on mac and think in a solution.
I create a new issue for your problem (see mentioned below).
Temporarily you can comment or remove the parts with color or color name.

I think if the script is running on your MacOs Catalina says that the @ajainvivek problem's has been solved.
Can you test for me, please? Just to make sure

Change this code:
```
IFS= read -sn1 key 2>/dev/null >&2
read -sn1 -t 0.0001 k1
read -sn1 -t 0.0001 k2
read -sn1 -t 0.0001 k3
```

For this:
```
IFS= read -sN1 key 2>/dev/null >&2
read -sN1 -t 0.0001 k1
read -sN1 -t 0.0001 k2
read -sN1 -t 0.0001 k3
```

I believe that if the script stops working, with this change, but works with the `master` branch code, then the first problem was solved

**Comment by andreysmyntyna** [comment]
@pedro-hs 

Hi,

I would suggest to utilise AWS to gain an access to a MacOS instance. 
https://aws.amazon.com/ru/blogs/aws/new-use-mac-instances-to-build-test-macos-ios-ipados-tvos-and-watchos-apps/

I believe that it will be a productive way to debug the script :-)

**Comment by pedro-hs** [comment]
Thanks @andreysmyntyna
